### PR TITLE
Fix disks cleanup over BM

### PR DESCRIPTION
**Pull request by prsurve**
Fixes: #4597 

Signed-off-by: Pratik Surve <prsurve@redhat.com>

### CODE REVIEW

**Comment by prsurve** [comment]
Verified Locally http://pastebin.test.redhat.com/980992

File: ocs_ci/deployment/baremetal.py [hunk 1]
```diff
@@ -706,6 +706,16 @@ def clean_disk():
...
 
     for worker in workers:
+        cmd = """lsblk --all --noheadings --output "KNAME,PKNAME,TYPE,MOUNTPOINT" --json"""
+        out = ocp_obj.exec_oc_debug_cmd(node=worker.name, cmd_list=[cmd])
+        disk_to_ignore_cleanup_raw = json.loads(str(out))
+        disk_to_ignore_cleanup_json = disk_to_ignore_cleanup_raw["blockdevices"]
+        for disk_to_ignore_cleanup in disk_to_ignore_cleanup_json:
+            if disk_to_ignore_cleanup["mountpoint"] == "/boot":
+                logger.info(
+                    f"Ignorning disk {disk_to_ignore_cleanup['pkname']} for cleanup"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:716]: maybe mention in the log message the reason why it's not being cleaned

File: ocs_ci/deployment/baremetal.py [hunk 2]
```diff
@@ -719,8 +729,9 @@ def clean_disk():
             lsblk_output = json.loads(str(out))
             lsblk_devices_to_clean = lsblk_output["blockdevices"]
             for device_to_clean in lsblk_devices_to_clean:
-                if not device_to_clean.get("children"):
-                    logger.info("Cleaning Disk")
+                if device_to_clean["name"] == str(selected_disk_to_ignore_cleanup):
+                    logger.info(f"Skipping disk cleanup for {device_to_clean['name']}")
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:733]: same

File: ocs_ci/deployment/baremetal.py (hunk 1 above)
Comment (prsurve) [reply ocs_ci/deployment/baremetal.py]: Done added the reason for skipping

File: ocs_ci/deployment/baremetal.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/deployment/baremetal.py]: Done added the reason for skipping

File: ocs_ci/deployment/baremetal.py [hunk 3]
```diff
@@ -706,6 +706,16 @@ def clean_disk():
...
+        cmd = """lsblk --all --noheadings --output "KNAME,PKNAME,TYPE,MOUNTPOINT" --json"""
+        out = ocp_obj.exec_oc_debug_cmd(node=worker.name, cmd_list=[cmd])
+        disk_to_ignore_cleanup_raw = json.loads(str(out))
+        disk_to_ignore_cleanup_json = disk_to_ignore_cleanup_raw["blockdevices"]
+        for disk_to_ignore_cleanup in disk_to_ignore_cleanup_json:
+            if disk_to_ignore_cleanup["mountpoint"] == "/boot":
+                logger.info(
+                    f"Ignorning disk {disk_to_ignore_cleanup['pkname']} for cleanup"
+                )
+                selected_disk_to_ignore_cleanup = disk_to_ignore_cleanup["pkname"]
```
Comment (vavuthu) [inline ocs_ci/deployment/baremetal.py:718]: can we break here since we will have single boot partition

Comment (prsurve) [reply ocs_ci/deployment/baremetal.py:718]: boot disk will always have `/boot` in part and that particular disk will be skipped

Comment (vavuthu) [reply ocs_ci/deployment/baremetal.py:718]: once you found boot disk, you can break the for loop

Comment (prsurve) [reply ocs_ci/deployment/baremetal.py:718]: yes can be done will try to use it

Comment (prsurve) [reply ocs_ci/deployment/baremetal.py:718]: Done

**Review by vavuthu — APPROVED** [review]

**Review by jimdevops19 — APPROVED** [review]

**Pull request merged**